app/controllers/download_controller.rb

```diff
@@ -25,8 +25,8 @@ def show
     response = check_type
     validate response
     respond_to do |format|
-      format.json { render json: flash, response: response }
-      format.html { render json: flash, response: response }
+      format.json { render json: [flash, download_file_path(response)], response: response }
+      format.html { render json: [flash, download_file_path(response)], response: response }
```

Comment:
change the response from the server so that we have the link to download the file in the response...this allows us to construct a download link to replace the link that was used to generate the file (in addition to the showing the flash message)